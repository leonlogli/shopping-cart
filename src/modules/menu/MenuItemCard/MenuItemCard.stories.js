import React from 'react'
import { text } from '@storybook/addon-knobs'

import MenuItemCard from '.'

export default {
  title: 'MenuItemCard',
}

export const Default = () => (
  <MenuItemCard
    title="Tacos Viande hachée"
    image="https://www.clipartmax.com/png/middle/29-295276_clipart-of-restaurant-breakfast-and-foods-transparent-hamburger-and-fries-png.png"
    price={27}
    onClick={() => console.log('clicked')}
    onAdd={() => console.log('add')}
  />
)

export const withDescription = () => (
  <MenuItemCard
    title="Tacos Viande hachée"
    description="Frites + Poissons"
    image="https://www.clipartmax.com/png/middle/29-295276_clipart-of-restaurant-breakfast-and-foods-transparent-hamburger-and-fries-png.png"
    price={27}
  />
)

export const withKnobs = () => (
  <MenuItemCard
    title="Tacos Viande hachée"
    restaurant={text('Restaurant', 'Galicia')}
    description={text('description')}
    image={text('image')}
    price={27}
  />
)
