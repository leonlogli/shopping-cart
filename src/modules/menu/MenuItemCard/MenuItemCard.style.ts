import styled from 'styled-components'
import Paper from '@material-ui/core/Paper'

import { Box, H6 } from '../../../components'

const Title = styled(H6)`
  margin-top: 16px;
`

const FooterBox = styled(Box)`
  display: flex;
  justify-content: space-between;
  width: 100%;
  align-items: center;

  > p {
    text-align: center;
    font-weight: 500;
    font-size: 15px;
  }
`

const MenuItemCardRoot = styled(Paper)`
  padding: 0 16px;
  display: flex;
  flex-direction: column;
  cursor: pointer;
`

export { Title, MenuItemCardRoot, FooterBox }
