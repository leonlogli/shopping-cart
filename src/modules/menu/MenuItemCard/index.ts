export * from './MenuItemCard'
export { default } from './MenuItemCard'
