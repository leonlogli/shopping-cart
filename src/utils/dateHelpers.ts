const durationFormat = (interval: number, type: string) => {
  return `il y a ${Math.floor(interval)} ${type}${
    Math.floor(interval) > 1 ? 's' : ''
  }`
}

const fromNow = (date: Date) => {
  const seconds = Math.floor(
    ((new Date() as any) - (new Date(date) as any)) / 1000
  )

  let interval = seconds / 31536000

  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'an')
  }

  interval = seconds / 2592000
  if (Math.floor(interval) >= 1) {
    return `il y a ${Math.floor(interval)} mois`
  }

  interval = seconds / 86400
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'jour')
  }

  interval = seconds / 3600
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'heure')
  }

  interval = seconds / 60
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'minute')
  }

  return `Ã  l'instant`
}

/**
 * Parse the given string time to a date.Time must have `hh:mm:ss` format
 * @param time time to parse
 * @param date date to which the time will be set once parsed. If null, today's date is used
 */
const parseTimeToDate = (time: string, date?: Date) => {
  const dateToSet = date || new Date()
  const timeParts = time.split(':').map((t) => Number(t) || 0)

  dateToSet.setHours(timeParts[0], timeParts[1], timeParts[2] || 0)

  return dateToSet
}

const formatShortDate = (date: any, locale?: string) => {
  return new Date(date).toLocaleDateString(locale, {
    day: 'numeric',
    month: 'short',
    year: 'numeric',
  })
}

export { fromNow, parseTimeToDate, formatShortDate }
