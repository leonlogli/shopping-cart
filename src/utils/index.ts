export * from './testHelpers'
export * from './util'
export * from './dateHelpers'
