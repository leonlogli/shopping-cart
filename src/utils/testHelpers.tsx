import React from 'react'
import { ThemeProvider } from 'styled-components'
import { mount, shallow } from 'enzyme'
import { defaultTheme } from '../styles'

const ThemeProviderWrapper = ({ children }: any) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
)

export const shallowWithTheme = (tree: any) =>
  shallow(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })

export const mountWithTheme = (tree: any) =>
  mount(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })
