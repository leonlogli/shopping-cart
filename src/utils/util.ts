const toUSD = (value: number) => {
  return `${value.toLocaleString('en-US', {
    minimumFractionDigits: 2,
  })} USD`
}

export { toUSD }
