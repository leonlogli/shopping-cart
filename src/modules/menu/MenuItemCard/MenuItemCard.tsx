import React, { MouseEvent } from 'react'
import AddCircleOutlineIcon from '@material-ui/icons/AddCircleOutline'
import IconButton from '@material-ui/core/IconButton'
import { PaperProps } from '@material-ui/core/Paper'

import { Box, RatioBox, Text } from '../../../components'

import { Title, MenuItemCardRoot, FooterBox } from './MenuItemCard.style'

import { toUSD } from '../../../utils'

export type MenuItemCardProps = PaperProps & {
  title: string
  description?: string
  image?: string | null
  price: number
  restaurant?: string
  onAdd?: (event: MouseEvent<HTMLButtonElement>) => void
}

const MenuItemCard = (props: MenuItemCardProps) => {
  const {
    title,
    price,
    description,
    restaurant,
    image,
    className,
    onClick,
    onAdd,
    ...other
  } = props

  const handleAddClick = (event: MouseEvent<HTMLElement>) => {
    const isAddButtonClick = event.currentTarget.tagName === 'BUTTON'

    if (onAdd && isAddButtonClick) {
      event.stopPropagation()
      onAdd(event as any)
    }
  }

  return (
    <MenuItemCardRoot
      className={`MenuItemCard ${className || ''}`}
      {...other}
      onClick={onClick}
    >
      <Box flexGrow={1}>
        {image && (
          <RatioBox ratio={55} marginTop={2}>
            <img alt={title} width="100%" height="100%" src={image} />
          </RatioBox>
        )}
        <Title className="Title">{title}</Title>
        {description && <Text>{description}</Text>}
        {restaurant && <Text>{restaurant}</Text>}
      </Box>
      <FooterBox>
        <Text>{toUSD(price)}</Text>
        <IconButton
          onClick={handleAddClick}
          aria-label="add item"
          edge="end"
          color="primary"
        >
          <AddCircleOutlineIcon />
        </IconButton>
      </FooterBox>
    </MenuItemCardRoot>
  )
}

export { MenuItemCard }
export default MenuItemCard
