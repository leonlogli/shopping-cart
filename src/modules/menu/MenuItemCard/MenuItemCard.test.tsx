import React from 'react'

import { mountWithTheme } from '../../../utils'

import MenuItemCard from './MenuItemCard'

describe('<MenuItemCard />', () => {
  const wrapper = mountWithTheme(
    <MenuItemCard
      title="Tacos"
      image="https://www.uir.ac.ma/upload/cbuilder/3e5aa8eb56fb64e0d9340fa21687e7693c14fe56.png"
      price={27}
    />
  )

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
