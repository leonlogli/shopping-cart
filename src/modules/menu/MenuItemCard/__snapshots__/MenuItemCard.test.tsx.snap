// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuItemCard /> should match a snapshot 1`] = `
<MenuItemCard
  image="https://www.uir.ac.ma/upload/cbuilder/3e5aa8eb56fb64e0d9340fa21687e7693c14fe56.png"
  price={27}
  title="Tacos"
>
  <Styled(WithStyles(ForwardRef(Paper)))
    className="MenuItemCard "
  >
    <WithStyles(ForwardRef(Paper))
      className="duKFOh MenuItemCard "
    >
      <ForwardRef(Paper)
        className="duKFOh MenuItemCard "
        classes={
          Object {
            "elevation0": "MuiPaper-elevation0",
            "elevation1": "MuiPaper-elevation1",
            "elevation10": "MuiPaper-elevation10",
            "elevation11": "MuiPaper-elevation11",
            "elevation12": "MuiPaper-elevation12",
            "elevation13": "MuiPaper-elevation13",
            "elevation14": "MuiPaper-elevation14",
            "elevation15": "MuiPaper-elevation15",
            "elevation16": "MuiPaper-elevation16",
            "elevation17": "MuiPaper-elevation17",
            "elevation18": "MuiPaper-elevation18",
            "elevation19": "MuiPaper-elevation19",
            "elevation2": "MuiPaper-elevation2",
            "elevation20": "MuiPaper-elevation20",
            "elevation21": "MuiPaper-elevation21",
            "elevation22": "MuiPaper-elevation22",
            "elevation23": "MuiPaper-elevation23",
            "elevation24": "MuiPaper-elevation24",
            "elevation3": "MuiPaper-elevation3",
            "elevation4": "MuiPaper-elevation4",
            "elevation5": "MuiPaper-elevation5",
            "elevation6": "MuiPaper-elevation6",
            "elevation7": "MuiPaper-elevation7",
            "elevation8": "MuiPaper-elevation8",
            "elevation9": "MuiPaper-elevation9",
            "outlined": "MuiPaper-outlined",
            "root": "MuiPaper-root",
            "rounded": "MuiPaper-rounded",
          }
        }
      >
        <div
          className="MuiPaper-root duKFOh MenuItemCard  MuiPaper-elevation1 MuiPaper-rounded"
        >
          <ForwardRef
            flexGrow={1}
          >
            <Styled(Styled(MuiBox))
              flexGrow={1}
            >
              <Styled(MuiBox)
                className="dUNeVP"
                flexGrow={1}
              >
                <div
                  className="MuiBox-root MuiBox-root-1 dUNeVP"
                >
                  <RatioBox
                    marginTop={2}
                    ratio={55}
                  >
                    <ForwardRef
                      marginTop={2}
                    >
                      <Styled(Styled(MuiBox))
                        marginTop={2}
                      >
                        <Styled(MuiBox)
                          className="dUNeVP"
                          marginTop={2}
                        >
                          <div
                            className="MuiBox-root MuiBox-root-2 dUNeVP"
                          >
                            <Styled(Component)
                              ratio={55}
                            >
                              <ForwardRef
                                className="eEgxxS"
                                ratio={55}
                              >
                                <Styled(Styled(MuiBox))
                                  className="eEgxxS"
                                  ratio={55}
                                >
                                  <Styled(MuiBox)
                                    className="dUNeVP eEgxxS"
                                    ratio={55}
                                  >
                                    <div
                                      className="MuiBox-root MuiBox-root-3 dUNeVP eEgxxS"
                                      ratio={55}
                                    >
                                      <Styled(Component)>
                                        <ForwardRef
                                          className="eMiRfY"
                                        >
                                          <Styled(Styled(MuiBox))
                                            className="eMiRfY"
                                          >
                                            <Styled(MuiBox)
                                              className="dUNeVP eMiRfY"
                                            >
                                              <div
                                                className="MuiBox-root MuiBox-root-4 dUNeVP eMiRfY"
                                              >
                                                <img
                                                  alt="Tacos"
                                                  height="100%"
                                                  src="https://www.uir.ac.ma/upload/cbuilder/3e5aa8eb56fb64e0d9340fa21687e7693c14fe56.png"
                                                  width="100%"
                                                />
                                              </div>
                                            </Styled(MuiBox)>
                                          </Styled(Styled(MuiBox))>
                                        </ForwardRef>
                                      </Styled(Component)>
                                    </div>
                                  </Styled(MuiBox)>
                                </Styled(Styled(MuiBox))>
                              </ForwardRef>
                            </Styled(Component)>
                          </div>
                        </Styled(MuiBox)>
                      </Styled(Styled(MuiBox))>
                    </ForwardRef>
                  </RatioBox>
                  <Styled(H6)
                    className="Title"
                  >
                    <H6
                      className="bSZpwu Title"
                    >
                      <Styled(WithStyles(ForwardRef(Typography)))
                        className="bSZpwu Title"
                        variant="h6"
                      >
                        <WithStyles(ForwardRef(Typography))
                          className="bSZpwu Title"
                          variant="h6"
                        >
                          <ForwardRef(Typography)
                            className="bSZpwu Title"
                            classes={
                              Object {
                                "alignCenter": "MuiTypography-alignCenter",
                                "alignJustify": "MuiTypography-alignJustify",
                                "alignLeft": "MuiTypography-alignLeft",
                                "alignRight": "MuiTypography-alignRight",
                                "body1": "MuiTypography-body1",
                                "body2": "MuiTypography-body2",
                                "button": "MuiTypography-button",
                                "caption": "MuiTypography-caption",
                                "colorError": "MuiTypography-colorError",
                                "colorInherit": "MuiTypography-colorInherit",
                                "colorPrimary": "MuiTypography-colorPrimary",
                                "colorSecondary": "MuiTypography-colorSecondary",
                                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                "displayBlock": "MuiTypography-displayBlock",
                                "displayInline": "MuiTypography-displayInline",
                                "gutterBottom": "MuiTypography-gutterBottom",
                                "h1": "MuiTypography-h1",
                                "h2": "MuiTypography-h2",
                                "h3": "MuiTypography-h3",
                                "h4": "MuiTypography-h4",
                                "h5": "MuiTypography-h5",
                                "h6": "MuiTypography-h6",
                                "noWrap": "MuiTypography-noWrap",
                                "overline": "MuiTypography-overline",
                                "paragraph": "MuiTypography-paragraph",
                                "root": "MuiTypography-root",
                                "srOnly": "MuiTypography-srOnly",
                                "subtitle1": "MuiTypography-subtitle1",
                                "subtitle2": "MuiTypography-subtitle2",
                              }
                            }
                            variant="h6"
                          >
                            <h6
                              className="MuiTypography-root bSZpwu Title MuiTypography-h6"
                            >
                              Tacos
                            </h6>
                          </ForwardRef(Typography)>
                        </WithStyles(ForwardRef(Typography))>
                      </Styled(WithStyles(ForwardRef(Typography)))>
                    </H6>
                  </Styled(H6)>
                </div>
              </Styled(MuiBox)>
            </Styled(Styled(MuiBox))>
          </ForwardRef>
          <Styled(Component)>
            <ForwardRef
              className="gWamZE"
            >
              <Styled(Styled(MuiBox))
                className="gWamZE"
              >
                <Styled(MuiBox)
                  className="dUNeVP gWamZE"
                >
                  <div
                    className="MuiBox-root MuiBox-root-5 dUNeVP gWamZE"
                  >
                    <Text>
                      <Styled(WithStyles(ForwardRef(Typography)))>
                        <WithStyles(ForwardRef(Typography))
                          className=""
                        >
                          <ForwardRef(Typography)
                            className=""
                            classes={
                              Object {
                                "alignCenter": "MuiTypography-alignCenter",
                                "alignJustify": "MuiTypography-alignJustify",
                                "alignLeft": "MuiTypography-alignLeft",
                                "alignRight": "MuiTypography-alignRight",
                                "body1": "MuiTypography-body1",
                                "body2": "MuiTypography-body2",
                                "button": "MuiTypography-button",
                                "caption": "MuiTypography-caption",
                                "colorError": "MuiTypography-colorError",
                                "colorInherit": "MuiTypography-colorInherit",
                                "colorPrimary": "MuiTypography-colorPrimary",
                                "colorSecondary": "MuiTypography-colorSecondary",
                                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                "displayBlock": "MuiTypography-displayBlock",
                                "displayInline": "MuiTypography-displayInline",
                                "gutterBottom": "MuiTypography-gutterBottom",
                                "h1": "MuiTypography-h1",
                                "h2": "MuiTypography-h2",
                                "h3": "MuiTypography-h3",
                                "h4": "MuiTypography-h4",
                                "h5": "MuiTypography-h5",
                                "h6": "MuiTypography-h6",
                                "noWrap": "MuiTypography-noWrap",
                                "overline": "MuiTypography-overline",
                                "paragraph": "MuiTypography-paragraph",
                                "root": "MuiTypography-root",
                                "srOnly": "MuiTypography-srOnly",
                                "subtitle1": "MuiTypography-subtitle1",
                                "subtitle2": "MuiTypography-subtitle2",
                              }
                            }
                          >
                            <p
                              className="MuiTypography-root MuiTypography-body1"
                            >
                              27.00 USD
                            </p>
                          </ForwardRef(Typography)>
                        </WithStyles(ForwardRef(Typography))>
                      </Styled(WithStyles(ForwardRef(Typography)))>
                    </Text>
                    <WithStyles(ForwardRef(IconButton))
                      aria-label="add item"
                      color="primary"
                      edge="end"
                      onClick={[Function]}
                    >
                      <ForwardRef(IconButton)
                        aria-label="add item"
                        classes={
                          Object {
                            "colorInherit": "MuiIconButton-colorInherit",
                            "colorPrimary": "MuiIconButton-colorPrimary",
                            "colorSecondary": "MuiIconButton-colorSecondary",
                            "disabled": "Mui-disabled",
                            "edgeEnd": "MuiIconButton-edgeEnd",
                            "edgeStart": "MuiIconButton-edgeStart",
                            "label": "MuiIconButton-label",
                            "root": "MuiIconButton-root",
                            "sizeSmall": "MuiIconButton-sizeSmall",
                          }
                        }
                        color="primary"
                        edge="end"
                        onClick={[Function]}
                      >
                        <WithStyles(ForwardRef(ButtonBase))
                          aria-label="add item"
                          centerRipple={true}
                          className="MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeEnd"
                          disabled={false}
                          focusRipple={true}
                          onClick={[Function]}
                        >
                          <ForwardRef(ButtonBase)
                            aria-label="add item"
                            centerRipple={true}
                            className="MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeEnd"
                            classes={
                              Object {
                                "disabled": "Mui-disabled",
                                "focusVisible": "Mui-focusVisible",
                                "root": "MuiButtonBase-root",
                              }
                            }
                            disabled={false}
                            focusRipple={true}
                            onClick={[Function]}
                          >
                            <button
                              aria-label="add item"
                              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeEnd"
                              disabled={false}
                              onBlur={[Function]}
                              onClick={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              tabIndex={0}
                              type="button"
                            >
                              <span
                                className="MuiIconButton-label"
                              >
                                <ForwardRef>
                                  <WithStyles(ForwardRef(SvgIcon))>
                                    <ForwardRef(SvgIcon)
                                      classes={
                                        Object {
                                          "colorAction": "MuiSvgIcon-colorAction",
                                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                                          "colorError": "MuiSvgIcon-colorError",
                                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                          "root": "MuiSvgIcon-root",
                                        }
                                      }
                                    >
                                      <svg
                                        aria-hidden={true}
                                        className="MuiSvgIcon-root"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                                        />
                                      </svg>
                                    </ForwardRef(SvgIcon)>
                                  </WithStyles(ForwardRef(SvgIcon))>
                                </ForwardRef>
                              </span>
                              <WithStyles(memo)
                                center={true}
                              >
                                <ForwardRef(TouchRipple)
                                  center={true}
                                  classes={
                                    Object {
                                      "child": "MuiTouchRipple-child",
                                      "childLeaving": "MuiTouchRipple-childLeaving",
                                      "childPulsate": "MuiTouchRipple-childPulsate",
                                      "ripple": "MuiTouchRipple-ripple",
                                      "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                      "rippleVisible": "MuiTouchRipple-rippleVisible",
                                      "root": "MuiTouchRipple-root",
                                    }
                                  }
                                >
                                  <span
                                    className="MuiTouchRipple-root"
                                  >
                                    <TransitionGroup
                                      childFactory={[Function]}
                                      component={null}
                                      exit={true}
                                    />
                                  </span>
                                </ForwardRef(TouchRipple)>
                              </WithStyles(memo)>
                            </button>
                          </ForwardRef(ButtonBase)>
                        </WithStyles(ForwardRef(ButtonBase))>
                      </ForwardRef(IconButton)>
                    </WithStyles(ForwardRef(IconButton))>
                  </div>
                </Styled(MuiBox)>
              </Styled(Styled(MuiBox))>
            </ForwardRef>
          </Styled(Component)>
        </div>
      </ForwardRef(Paper)>
    </WithStyles(ForwardRef(Paper))>
  </Styled(WithStyles(ForwardRef(Paper)))>
</MenuItemCard>
`;
